pkg/reporter/table_reporter.go

```diff
@@ -17,22 +18,23 @@ type TableReporter struct {
 	terminalWidth int
 }
 
-func NewTableReporter(stdout io.Writer, stderr io.Writer, markdown bool, terminalWidth int) *TableReporter {
+func NewTableReporter(stdout io.Writer, stderr io.Writer, level VerbosityLevel, markdown bool, terminalWidth int) *TableReporter {
 	return &TableReporter{
-		stdout:          stdout,
-		stderr:          stderr,
-		hasPrintedError: false,
-		markdown:        markdown,
-		terminalWidth:   terminalWidth,
+		VerbosityLevelPrinter: NewVerbosityLevelPrinter(level),
```

Comment:
I'm unsure if I should use an embedded struct; `VerbosityLevelPrinter` is just a wrapper over `fmt.Fprintf(...)` , where we check to make sure we can print at the provided level before calling it. We could alternatively make it a field or find some other approach to avoid having to duplicate this check across all reporters

Reply:
Also, is there any value in writing unit tests for these new `Reporter` methods? If so, would it be better to create a *_test.go for each `Reporter` implementation to test them separately. Or, could I create a reporter_test.go and just test all the implementations here (I guess the value of these tests would be to make sure they are printing in their respective level e.g. a `Verbosef()` implementation is actually printing in the `VerboseLevel`)?

Reply:
> I'm unsure if I should use an embedded struct; VerbosityLevelPrinter is just a wrapper over fmt.Fprintf(...) , where we check to make sure we can print at the provided level before calling it. We could alternatively make it a field or find some other approach to avoid having to duplicate this check across all who implement Reporter

Hmm I don't think it's worth the complexity of having the embeded struct, it makes the code a bit harder to understand.

Personally I think there's not that much repetition to have the if statement in every print statement, and makes it very clear what is happening. 
```
func (r *TableReporter) Warnf(format string, a ...any) {
    if WarnLevel <= r.level {
        fmt.Fprintf(writer, format, a...)
    }
}
```

This part of the code also doesn't have to change if we want to add additional levels in the future.

Reply:
> Also, is there any value in writing unit tests for these new Reporter methods? If so, would it be better to create a *_test.go for each Reporter implementation to test them separately. Or, could I create a reporter_test.go and just test all the implementations here (I guess the value of these tests would be to make sure they are printing in their respective level e.g. a Verbosef() implementation is actually printing in the VerboseLevel)?

Let's have them in separate _test.go files, I think it's worth testing now that there's additional logic here.